gregBorchers/main.js

```diff
@@ -0,0 +1,71 @@
+// Your card and deque factory code should be loaded into the browser before this file
+// by the main.html container file.
+// At this point, both makeCard and makeDeque should be defined.
+
+
+// 2b:
+// make a deque instance to store a full deck of cards:
+var myCardFactory = new cardFactory();
+cardFactory.makeFullSet(myCardFactory);
+
+var myDequeFactory = new dequeFactory();
+
+var myDeckOfCards = myDequeFactory.makeDeque(myCardFactory.fullSet);
+
+console.log("--------------------------------BEFORE Sort by suit");
+for (i=0; i< myDeckOfCards.deque.length; i++ ) {
+	console.log( "card " + i + " is " + myDeckOfCards.deque[i].name() );
+}
```

Comment:
Remember that this should also sort by rank within suit! Because the sort function compares things in a fairly inscrutable fashion, at present you're left with each suit fairly scrambled, which is why your King of Diamonds test is failing.